chef/automate
# Jamie/1192 manual deletion of services

### :nut_and_bolt: Description: What code changed, and why?
This branch updates the applications page so that users have the ability to manually delete services from the UI.

### :chains: Related Resources
Adds UI support for https://github.com/chef/automate/issues/1192

### :+1: Definition of Done
Users can successfully delete services from the UI.  Upon success, users are shown a success notification.  Upon failure, users are shown a failure notification.

### :athletic_shoe: How to Build and Test the Change
cd into automate: `build components/automate-ui-devproxy && start_all_services`
cd into automate/components/automate-ui: `make serve`

To Test:

1. Navigate to https://a2-dev.test/applications/service-groups
2. Select a service group in the service group table
3. On the right-side panel, select one service instance card with the checkbox, see the bulk action bar status update as described in the Acceptance Criteria
4. Select one more service instance card
5. Click on the Select All button, see that all the service instance on the current page is selected.
6. Navigate to another page, see the Select All button is set to unselected, and the Delete Service button changed to disabled. Toggle the Select All checkbox, see that all services on the page are selected. Toggle the button again, see all services are deselected.
7. Select a random number of services on the page, click on the Delete Service button.
8. In the confirmation modal, click on the Cancel button, see the modal dismissed, and the previous selection state persists.
9. Click on the Delete Service button again, this time, confirm the deletion in the modal.
10. See that the right-side panel changes to a deleting state.
11. Simulate two situations where the delete operation is failed and successful, show the two banners as described in the Acceptance Criteria.

### :white_check_mark: Checklist


### :camera: Screenshots, if applicable
![Manual Deletion of Services](https://user-images.githubusercontent.com/16737484/87727830-a7974b00-c776-11ea-8ff0-53c89476bd8d.gif)
